### PR TITLE
feat: Get element by id

### PR DESCRIPTION
**Pull request by tedconn**
This PR adds `getElementById(id: string, context?: Node)`.

If the provided context is not a `Document`, `ShadowRoot` or `HTMLDocument` it will throw a TypeError

This PR also increases the coverage to 100% which allows us to set the threshold for 100%! 🥳

### CODE REVIEW

**Comment by nolanlawson** [comment]
> The `document` in tests always refers to the karma document, and `getElementById` doesn't penetrate that iframe where the shadow DOM is created.

We can always do `iframe.contentWindow.document`, right?

If the iframe is a hassle then we can remove it for these particular tests, but it seems to me we should be able to just grab the iframe's document.

File: src/index.js [hunk 1]
```diff
@@ -80,6 +80,7 @@ function getParent (element) {
   }
   // if an element is inside the shadow DOM, break outside of it
   const rootNode = element.getRootNode()
+  /* istanbul ignore else */
```
Comment (nolanlawson) [inline src/index.js:83]: Coverage fixes are orthogonal to `getElementById` and should probably go in another PR.

File: src/index.js [hunk 2]
```diff
@@ -302,10 +316,19 @@ function getElementsByClassName (classNames, context = document) {
   return getMatchingElementsByClassName(elementIterator, classNamesSplit, context)
 }
 
+function getElementById (id, context = document) {
+  if (context.constructor.name !== 'Document' && context.constructor.name !== 'HTMLDocument' && context.constructor.name !== 'ShadowRoot') {
+    throw new TypeError('Provided context must be of type Document or ShadowRoot')
```
Comment (nolanlawson) [inline src/index.js:321]: Awesome

**Comment by nolanlawson** [comment]
^ Merge conflicts, other than that LGTM

File: src/index.js [hunk 3]
```diff
@@ -248,6 +251,17 @@ function getMatchingElementsByClassName (elementIterator, classNames) {
   return results
 }
 
+function getMatchingElementById (elementIterator, id) {
+  let element
+  while ((element = elementIterator.next())) {
```
Comment (blittle) [inline src/index.js:256]: I love the usage of iterators. Too bad we can't (I assume) use generators or `for of` due to IE11 support.

Comment (nolanlawson) [reply src/index.js:256]: We are compiling it down to ES5 using Bublé. But yeah generators tend to add a lot of overhead in that case.

File: src/index.js (hunk 1 above)
Comment (tedconn) [reply src/index.js]: @nolanlawson I removed the coverage fixes from this PR

**Review by nolanlawson — APPROVED** [review]
